use-cases/virgo/trainer.py

```diff
@@ -385,7 +379,233 @@ def train(self):
 
             # Report training metrics of last epoch to Ray
             train.report({"loss": np.mean(val_loss), "train_loss": np.mean(epoch_loss)})
+
+        return loss_plot, val_loss_plot, acc_plot, val_acc_plot
+
+
+class RayNoiseGeneratorTrainer(RayTorchTrainer):
+    def __init__(
+        self,
+        config: Dict,
+        strategy: Optional[Literal["ddp", "deepspeed"]] = "ddp",
+        name: Optional[str] = None,
+        logger: Optional[Logger] = None,
+    ) -> None:
+        super().__init__(config=config, strategy=strategy, name=name, logger=logger)
+
+    def create_model_loss_optimizer(self) -> None:
+        # Select generator
+        generator = self.training_config["generator"]
```

Comment:
Just want to double check since I've seen you check these things before: Can you be sure that `self.training_config` contains "generator" here? 😁

Reply:
Hmm well no not exactly, so far this has to be defined in the config.yaml file, otherwise there will be an error. This is one of the reasons I will change the structure with the dictionary object in a future PR, for type-checking and checking that all fields are set.